core/search/v5.py

```diff
@@ -0,0 +1,299 @@
+    WORK_LAST_UPDATE_SCRIPT = """
+double champion = -1;
+// Start off by looking at the work's last update time.
+for (candidate in doc['last_update_time']) {
+    if (champion == -1 || candidate > champion) { champion = candidate; }
+}
+if (params.collection_ids != null && params.collection_ids.length > 0) {
+    // Iterate over all licensepools looking for a pool in a collection
+    // relevant to this filter. When one is found, check its
+    // availability time to see if it's later than the last update time.
+    for (licensepool in params._source.licensepools) {
+        if (!params.collection_ids.contains(licensepool['collection_id'])) { continue; }
+        double candidate = licensepool['availability_time'];
+        if (champion == -1 || candidate > champion) { champion = candidate; }
+    }
+}
+if (params.list_ids != null && params.list_ids.length > 0) {
+
+    // Iterate over all customlists looking for a list relevant to
+    // this filter. When one is found, check the previous work's first
+    // appearance on that list to see if it's later than the last
+    // update time.
+    for (customlist in params._source.customlists) {
+        if (!params.list_ids.contains(customlist['list_id'])) { continue; }
+        double candidate = customlist['first_appearance'];
+        if (champion == -1 || candidate > champion) { champion = candidate; }
+    }
+}
+
+return champion;
+"""
+
+    # Use regular expressions to normalized values in sortable fields.
+    # These regexes are applied in order; that way "H. G. Wells"
+    # becomes "H G Wells" becomes "HG Wells".
+    CHAR_FILTERS = {
+        "remove_apostrophes": dict(
+            type="pattern_replace",
+            pattern="'",
+            replacement="",
+        )
+    }
+
+    AUTHOR_CHAR_FILTER_NAMES = []
+    for name, pattern, replacement in [
+        # The special author name "[Unknown]" should sort after everything
+        # else. REPLACEMENT CHARACTER is the final valid Unicode character.
+        ("unknown_author", r"\[Unknown\]", "\N{REPLACEMENT CHARACTER}"),
+        # Works by a given primary author should be secondarily sorted
+        # by title, not by the other contributors.
+        ("primary_author_only", r"\s+;.*", ""),
+        # Remove parentheticals (e.g. the full name of someone who
+        # goes by initials).
+        ("strip_parentheticals", r"\s+\([^)]+\)", ""),
+        # Remove periods from consideration.
+        ("strip_periods", r"\.", ""),
+        # Collapse spaces for people whose sort names end with initials.
+        ("collapse_three_initials", r" ([A-Z]) ([A-Z]) ([A-Z])$", " $1$2$3"),
+        ("collapse_two_initials", r" ([A-Z]) ([A-Z])$", " $1$2"),
+    ]:
+        normalizer = dict(
+            type="pattern_replace", pattern=pattern, replacement=replacement
+        )
+        CHAR_FILTERS[name] = normalizer
+        AUTHOR_CHAR_FILTER_NAMES.append(name)
+
+    def __init__(self):
+        super().__init__(5)
```

Comment:
I changed this up to not pass the `SEARCH_VERSION` as an argument, since it is expected they all define the vairable.
The parent class just does this instead.

```python
...
def __init__():
    self._version = self.SEARCH_VERSION
...
```

Reply:
@io7m Yeah, that's a good point. I'd prefer a mandatory keyword here, as well. The keyword can make it more clear what this value is and the superclass can ensure that a value was passed.

Reply:
How about I raise an error if it is not defined correctly.